src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -60,25 +70,26 @@ public GpioController(PinNumberingScheme numberingScheme, GpioDriver driver)
         /// <summary>
         /// The number of pins provided by the controller.
         /// </summary>
-        public int PinCount => _driver.PinCount;
+        public virtual int PinCount => _driver.PinCount;
 
         /// <summary>
         /// Gets the logical pin number in the controller's numbering scheme.
         /// </summary>
-        /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
+        /// <param name="pinNumber">The pin number</param>
+        /// <param name="givenScheme">The scheme for the pin number</param>
         /// <returns>The logical pin number in the controller's numbering scheme.</returns>
-        private int GetLogicalPinNumber(int pinNumber)
+        protected virtual int GetLogicalPinNumber(int pinNumber, PinNumberingScheme givenScheme)
         {
-            return (NumberingScheme == PinNumberingScheme.Logical) ? pinNumber : _driver.ConvertPinNumberToLogicalNumberingScheme(pinNumber);
+            return _driver.ConvertPinNumberToLogicalNumberingScheme(pinNumber);
         }
 
         /// <summary>
         /// Opens a pin in order for it to be ready to use.
         /// </summary>
         /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
-        public void OpenPin(int pinNumber)
+        public virtual void OpenPin(int pinNumber)
```

Comment:
@Ellerbach @Ellerbach if we ship this in this release then likely we won't be able to relatively safely change return type as previously source compatible change can now be overriden by someone and now it won't be source compatible